src/com/dd/levels/DungeonMap.java

```diff
@@ -215,38 +176,47 @@ public boolean isOutOfBounds(MapPosition pos) {
 				|| pos.getY() < 0
 				|| pos.getY() > rooms[pos.getY()].length - 1;
 	}
+	
+	public MapPosition randPos(){
+		int yStart = rand.nextInt(10);
+		int xStart = rand.nextInt(10);
+		return new MapPosition(xStart, yStart);
+	}
+	
+	public void setRoom(Room room, MapPosition position) {
+		rooms[position.getY()][position.getX()] = room;
+	}
 
-	public void generateDungeon() {
+	public void generateDungeon(Room[][] rooms) {
```

Comment:
There's no reason to have rooms be a parameter. It doesn't make sense since dungeon map has the rooms.